vcd/resource_vcd_catalog_item.go

```diff
@@ -79,37 +87,66 @@ func resourceVcdCatalogItem() *schema.Resource {
 	}
 }
 
-func resourceVcdCatalogItemCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceVcdCatalogItemCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	log.Printf("[TRACE] Catalog item creation initiated")
 
 	vcdClient := meta.(*VCDClient)
 
 	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
 	if err != nil {
-		return fmt.Errorf(errorRetrievingOrg, err)
+		return diag.Errorf(errorRetrievingOrg, err)
 	}
 
 	catalogName := d.Get("catalog").(string)
 	catalog, err := adminOrg.GetCatalogByName(catalogName, false)
 	if err != nil {
 		log.Printf("[DEBUG] Error finding Catalog: %#v", err)
-		return fmt.Errorf("error finding Catalog: %#v", err)
+		return diag.Errorf("error finding Catalog: %#v", err)
```

Comment:
A good opportunity to cleanup this error. No need for `%#v` here

```suggestion
		return diag.Errorf("error finding Catalog: %s", err)
```

Reply:
changed in all places where could, still left in project